From f1fd024f6e9d02ec3d36226caf386e6be499135c Mon Sep 17 00:00:00 2001
From: Tao Ren <taoren@meta.com>
Date: Fri, 28 Oct 2022 11:50:15 -0700
Subject: [PATCH] ARM: dts: aspeed: Add Facebook janga BMC

Add initial version of device tree for Facebook janga (AST2600) BMC.

janga is Facebook's next generation switch platform with an AST2600
BMC integrated for health monitoring purpose.
---
 .../dts/aspeed-bmc-facebook-janga.dts     | 96 +++++++++++++++++++
 1 file changed, 96 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-facebook-janga.dts

diff --git a/arch/arm/boot/dts/aspeed-bmc-facebook-janga.dts b/arch/arm/boot/dts/aspeed-bmc-facebook-janga.dts
new file mode 100644
index 000000000000..b9c2edc22fe7
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-facebook-janga.dts
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: GPL-2.0+
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+/dts-v1/;
+
+#include "ast2600-facebook-netbmc-common.dtsi"
+
+/ {
+	model = "Facebook janga BMC";
+	compatible = "facebook,janga-bmc", "aspeed,ast2600";
+
+	chosen {
+		stdout-path = &uart1;
+	};
+
+	spi_gpio: spi-gpio {
+		num-chipselects = <1>;
+		cs-gpios = <&gpio0 ASPEED_GPIO(X, 0) GPIO_ACTIVE_LOW>;
+	};
+};
+
+&peci0 {
+	status = "okay";
+
+	peci-client@30 {
+		compatible = "intel,peci-client";
+		reg = <0x30>;
+	};
+};
+
+&mdio1 {
+	status = "okay";
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&lpc_ctrl {
+	status = "okay";
+	flash = <&spi1>;
+};
+
+&kcs1 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca0>;
+};
+
+&kcs2 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca8>;
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca2>;
+};
+
+/* JTAG1 will be use for COMe FPGA upgrading */
+&jtag1 {
+	status = "okay";
+};
+
+/* SPI1 will be use for BIOS, IOB_FPGA flash upgrading */
+&spi1 {
+	status = "okay";
+
+	reg = < 0x1e630000 0xc4
+			0x30000000 0x2000000 >;
+
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "spi1.0";
+		spi-rx-bus-width = <1>;
+		spi-max-frequency = <5000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash1@0 {
+				reg = <0x0 0x2000000>;
+				label = "spi-flash";
+			};
+		};
+	};
+};
-- 
2.39.3

